src/rocker/core.py

```diff
@@ -109,10 +132,69 @@ def extend_cli_parser(self, parser, default_args={}):
 
 
     def get_active_extensions(self, cli_args):
-        active_extensions = [e() for e in self.available_plugins.values() if e.check_args_for_activation(cli_args) and e.get_name() not in cli_args['extension_blacklist']]
-        active_extensions.sort(key=lambda e:e.get_name().startswith('user'))
-        return active_extensions
+        """
+        Checks for missing dependencies (specified by each extension's
+        required_extensions() method) and additionally sorts them.
+        """
+        active_extensions = {
+            name: cls for name, cls in self.available_plugins.items()
+            if cls.check_args_for_activation(cli_args) and cls.get_name() not in cli_args['extension_blacklist']
+        }
+        names = set(active_extensions.keys())
+        for name, cls in active_extensions.items():
+            if not cls.required_extensions().issubset(names):
+                raise RequiredExtensionMissingError(f"extension '{name}' is missing required extensions {list(cls.required_extensions())}")
```

Comment:
I took a stab at handling this nicely. 

One thing I noticed is that the current implementation allows for blacklisted extensions to silently fail. Is that desired behavior? I could add a message or error out similarly to how I'm handling that for the implicit required extensions

Reply:
I went ahead and updated the blacklist handling so that it alerts the user if they blacklist something that they also specified to be loaded. Feel free to remove the last commit if you don't like that behavior.